src/player.cpp

```diff
@@ -12251,26 +12251,36 @@ int player::encumb( body_part bp ) const
 int player::encumb(body_part bp, double &layers, int &armorenc) const
 {
     int ret = 0;
-    int layer[MAX_CLOTHING_LAYER] = { };
+    // First is the total of encumbrance on the given layer,
+    // and second is the highest encumbrance of any one item on the layer.
+    std::array<std::pair<int, int>, MAX_CLOTHING_LAYER> layer = {{{0,0},{0,0},{0,0},{0,0},{0,0}}};
```

Comment:
> std::array<std::pair<int, int>, MAX_CLOTHING_LAYER> layer(); 
> Did not do what I wanted

Yes, because it's a function declaration, everyone can see that (-;

However, what about this:
```C++
auto layer = std::array<std::pair<int, int>, MAX_CLOTHING_LAYER>();
// equivalent, but repeats the type:
std::array<std::pair<int, int>, MAX_CLOTHING_LAYER> layer = std::array<std::pair<int, int>, MAX_CLOTHING_LAYER>();
```
It uses default initialization, lists the type only once and thanks to copy copy elision, the array is created only once.

Oh, and Herb Sutter recommends that style anyway, http://herbsutter.com/2013/08/12/gotw-94-solution-aaa-style-almost-always-auto/